server/src/main/java/org/candlepin/dto/api/v1/ConsumerDTO.java

```diff
@@ -0,0 +1,1030 @@
+     */
+    private static class ReleaseVersionWrapSerializer extends SingleValueWrapSerializer {
+        public ReleaseVersionWrapSerializer() {
+            super("releaseVer");
+        }
+    }
+
+    /**
+     * Deserialization utility class for for unwrapping the 'releaseVer' field from a JSON object.
+     */
+    private static class ReleaseVersionWrapDeserializer extends SingleValueWrapDeserializer {
+        public ReleaseVersionWrapDeserializer() {
+            super("releaseVer");
+        }
+    }
+
+    /**
+     * Initializes a new ConsumerDTO instance with null values.
+     */
+    public ConsumerDTO() {
+        // Intentionally left blank
+    }
+
+    /**
+     * Initializes a new ConsumerDTO instance which is a shallow copy of the provided
+     * source entity.
+     *
+     * @param source
+     *  The source entity to copy
+     */
+    public ConsumerDTO(ConsumerDTO source) {
+        super(source);
+    }
+
+    /**
+     * Initializes a new ConsumerDTO. convenience constructor for tests.
+     * @param name the consumer name to set on the consumer
+     * @param userName the username to set on the consumer
+     * @param owner the owner to set on the consumer
+     * @param type the consumer type to set on the consumer
+     */
+    public ConsumerDTO(String name, String userName, OwnerDTO owner, ConsumerTypeDTO type) {
+        this();
+
+        this.name = name;
+        this.username = userName;
+        this.owner = owner;
+        this.type = type;
+        this.autoheal = true;
+        this.serviceLevel = "";
+        this.entitlementCount = 0L;
+        this.facts = new HashMap<String, String>();
+        this.installedProducts = new HashSet<ConsumerInstalledProductDTO>();
+        this.guestIds = new ArrayList<GuestIdDTO>();
+    }
+
+    /**
+     * Retrieves the id field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    @HateoasInclude
+    public String getId() {
+        return this.id;
+    }
+
+    /**
+     * Sets the id to set on this ConsumerDTO object.
+     *
+     * @param id the id to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Retrieves the uuid field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    @HateoasInclude
+    public String getUuid() {
+        return this.uuid;
+    }
+
+    /**
+     * Sets the uuid to set on this ConsumerDTO object.
+     *
+     * @param uuid the id to set on this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
+     */
+    public ConsumerDTO setUuid(String uuid) {
+        this.uuid = uuid;
+        return this;
+    }
+
+    /**
+     * Retrieves the name field of this ConsumerDTO object.
+     *
+     * @return a reference to this DTO object.
```

Comment:
Wrong return type comment. I won't add more comments for this, but there are quite a few methods (mostly getters) with this comment in this class that don't actually return a ConsumerDTO.